mock/service/controller.go

```diff
@@ -166,6 +166,11 @@ func (s *service) ControllerPublishVolume(
 		}, nil
 	}
 
+	// Check attach limit before publishing only if attach limit is set.
+	if int(s.config.AttachLimit) > 0 && s.getAttachCount(devPathKey) >= s.config.AttachLimit {
+		return nil, status.Errorf(codes.ResourceExhausted, "Cannot attach any more volumes to this node")
```

Comment:
Why are you casting `s.config.AttachLimit` here?

If I remember correctly, `0` is an untyped integer in Go, it should compare fine to both int64 and normal int.

Reply:
Thanks! I wasn't aware of that. Thought casting would be better.
Updated now.